### ISSUE TITLE
Tracking Issue for const Derefs

### ISSUE THREAD

**Issue by fee1-dead** [rust-lang/rust#88955]
Feature gate: `#![feature(const_deref)]`

This is a tracking issue for implementing `const Deref{,Mut}` in the standard library.

### Public API

```rust
// alloc::borrow

impl<B: ?Sized + ToOwned> const Deref for Cow<'_, B>
where
    B::Owned: ~const Borrow<B> {
    type Target = B;
}

// core::mem::manually_drop
impl<T: ?Sized> const Deref for ManuallyDrop<T> {
    type Target = T;
}
impl<T: ?Sized> const DerefMut for ManuallyDrop<T> {
    type Target = T;
}

// core::ops::deref
impl<T: ?Sized> const Deref for &mut T {
    type Target = T;
}
impl<T: ?Sized> const Deref for &T {
    type Target = T;
}
```

### Steps / History

- [x] Implementation: #88887
- [ ] Final comment period (FCP)
- [ ] Stabilization PR

### Unresolved Questions

- None yet.

**Comment by RalfJung** [comment]
This got entirely removed in https://github.com/rust-lang/rust/pull/110393.

**Comment by npmccallum** [comment]
I think this issue needs to be reopened. It is the only tracking issue for `const_deref` and there's definitely `const_deref` code in master.

**Comment by RalfJung** [comment]
No, the new implementation should use a new tracking issue.

**Comment by npmccallum** [comment]
@RalfJung my point is that the new implementation is already merged and already points to this issue.

**Comment by clarfonthey** [comment]
Yes, there are issues with the tracking issues for these traits and that is currently being fixed by #145279, which uses #143773 as its new tracking issue.

I'm currently not including these methods, but I can add those to my PR.
